crates/sui-sdk/examples/utils/example_utils.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use reqwest::Client;
+use serde_json::json;
+use shared_crypto::intent::Intent;
+use sui_config::{sui_config_dir, SUI_KEYSTORE_FILENAME};
+use sui_keys::keystore::{AccountKeystore, FileBasedKeystore};
+
+use crate::utils::wallet::retrieve_wallet;
+use futures::StreamExt;
+use std::future;
+use sui_sdk::types::{
+    base_types::{ObjectID, SuiAddress},
+    digests::TransactionDigest,
+    gas_coin::GasCoin,
+    quorum_driver_types::ExecuteTransactionRequestType,
+    transaction::{Transaction, TransactionData},
+};
+use sui_sdk::{rpc_types::SuiTransactionBlockResponseOptions, SuiClient, SuiClientBuilder};
+
+// const SUI_FAUCET: &str = "https://faucet.devnet.sui.io/gas"; // devnet faucet
+
+const SUI_FAUCET: &str = "https://faucet.testnet.sui.io/v1/gas"; // testnet faucet
+
+// if you use the sui-test-validator and use the local network; if it does not work, try with port 5003.
+// const SUI_FAUCET: &str = "http://127.0.0.1:9123/gas";
+
+/// Return a sui client to interact with the APIs,
+/// the active address, and another address that can be used to send stuff to,
+/// and the object id of the a coin owned by the active address that has at least 100_000_000 MIST on it.
+pub(crate) async fn setup_for_write(
+) -> Result<(SuiClient, SuiAddress, SuiAddress, ObjectID), anyhow::Error> {
+    let (client, active_address) = setup_for_read().await?;
+    let wallet = retrieve_wallet().await?;
+    let addresses = wallet.get_addresses();
+    let addresses = addresses
+        .into_iter()
+        .filter(|address| address != &active_address)
+        .collect::<Vec<_>>();
+    let recipient = addresses
+        .get(0)
+        .expect("Cannot get the recipient address needed for writing operations. Aborting");
+
+    let coin_obj_id = coin_with_sui(&client, active_address).await?;
+    Ok((client, active_address, *recipient, coin_obj_id))
+}
+
+/// Return a sui client to interact with the APIs, and the active address.
+pub(crate) async fn setup_for_read() -> Result<(SuiClient, SuiAddress), anyhow::Error> {
+    let client = SuiClientBuilder::default().build_testnet().await?;
+    println!("Sui testnet version is: {}", client.api_version());
+    let mut wallet = retrieve_wallet().await?;
+    assert!(wallet.get_addresses().len() >= 2);
+    let active_address = wallet.active_address()?;
+
+    Ok((client, active_address))
+}
+
+/// Return the coin object ID with at least 100_000_000 MIST owned by the provided active address.
+pub(crate) async fn coin_with_sui(
+    client: &SuiClient,
+    active_address: SuiAddress,
+) -> Result<ObjectID, anyhow::Error> {
+    let coin_type = "0x2::sui::SUI".to_string();
+
+    let coins = client
+        .coin_read_api()
+        .get_coins_stream(active_address, Some(coin_type.clone()))
+        .filter(|coin| future::ready(coin.balance >= 100_000_000))
+        .collect::<Vec<_>>()
+        .await;
```

Comment:
Note that my original suggestion was to use `get_balance` to figure out if the balance of all coins on the active address exceeded the amount you wanted.  The current approach works, but you don't need to enforce that all the balance is in one coin to use it, because of gas smashing.

Reply:
Also, I would expect their to be an API that allows you to await for the next element in a stream rather than collect everything in a vector and then get the first element.